lib/tasks/pulsearch.rake

```diff
@@ -31,6 +32,17 @@ namespace :pulsearch do
     end
   end
 
+  desc 'Generate a robots.txt file'
+  task :robots_txt do |_t, args|
+    file_path = args[:file_path] || Rails.root.join('public', 'robots.txt')
+    robots = RobotsGeneratorService.new(path: file_path, disallowed_paths: Rails.configuration.robots.disallowed_paths)
+    robots.insert_group(user_agent: '*')
+    robots.insert_crawl_delay(10)
+    robots.insert_sitemap(Rails.configuration.robots.sitemap_url)
+    robots.generate
+    robots.write
+  end
```

Comment:
I'm curious about how you see this working going forward.  Do you think we should periodically regenerate our `robots.txt` and compare to the checked-in static version? Or is this code that was useful to generate the original static version, that we don't need to maintain?

Reply:
> is this code that was useful to generate the original static version, that we don't need to maintain?
I felt that this was the preferred approach, with the hope that any adjustment to the Routes for the App would require the config. being adjusted (please see https://github.com/pulibrary/orangelight/pull/1050/files#diff-b29c2e1a89929db60557b9de48f7b633).  At the time that this PR was issued, I could not determine a straightforward approach to accessing all Rails routes and finding some more automated way to blacklist certain patterns, so unfortunately, the robots.txt rules will require some curation.